common/v2/features/Dashboard/components/WalletBreakdown/helpers.ts

```diff
@@ -0,0 +1,19 @@
+export const calculateShownIndex = (
+  numberOfPieChartSlices: number,
+  selectedIndex: number
+): number => {
+  // selectedIndex is the index of the asset balance element that is being hovered over.
+  // chartBalancesLength is the number of slices in the WalletBreakdown pie chart.
+  if (numberOfPieChartSlices > selectedIndex && selectedIndex !== -1) {
+    // This will show the relevant onHovered slice.
+    return selectedIndex;
+  } else {
+    if (numberOfPieChartSlices <= selectedIndex) {
+      // This will show the last element in the chartBalances array, which is the combined "other tokens" slice.
+      return numberOfPieChartSlices - 1;
+    } else {
+      // Default to showing the largest slice in the pie chart.
+      return 0;
+    }
```

Comment:
My issue was not with the ternary but with the nesting ;-)
```
if (numberOfPieChartSlices > selectedIndex && selectedIndex !== -1) { 
 return selectedIndex
} else if (numberOfPieChartSlices <= selectedIndex) {
 return numberOfPieChartSlices - 1;
} else {
 return 0;
}
```

or 

```
if (numberOfPieChartSlices > selectedIndex && selectedIndex !== -1) return selectedIndex;
if (numberOfPieChartSlices <= selectedIndex) return umberOfPieChartSlices - 1;
return 0;
```

Reply:
will fix this today, then merge in c: